dvc/stage/cache.py

```diff
@@ -197,7 +197,7 @@ def _transfer(func, from_remote, to_remote):
             return []
 
         for src in from_remote.walk_files(runs):
-            rel = src.relative_to(from_remote.path_info)
+            rel = PathInfo(os.path.relpath(src, from_remote.path_info))
```

Comment:
I think this change is no longer needed after the fix in `path_info.py`, @musa-atlihan can you try reverting this and then confirm that `pull --run-cache` works as expected?